homeassistant/components/surepetcare/__init__.py

```diff
@@ -99,12 +100,12 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             entry,
             hass,
         )
-    except SurePetcareAuthenticationError:
+    except SurePetcareAuthenticationError as error:
         _LOGGER.error("Unable to connect to surepetcare.io: Wrong credentials!")
-        return False
+        raise ConfigEntryAuthFailed from error
     except SurePetcareError as error:
         _LOGGER.error("Unable to connect to surepetcare.io: Wrong %s!", error)
```

Comment:
```suggestion
```

Reply:
We don't want it to spam the log every time. `ConfigEntryNotReady` will log once and downgrade to debug.